File: server/order-service/src/main/java/com/sudo248/orderservice/repository/entity/order/Shipment.java
package com.sudo248.orderservice.repository.entity.order;

import com.sudo248.orderservice.external.GHNService;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Embeddable;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Embeddable
public class Shipment {
    @Column(name = "shipment_type")
    private ShipmentType shipmentType = GHNService.DEFAULT_SHIPMENT_TYPE;
    @Column(name = "delivery_time")
    private int deliveryTime = 0;
    @Column(name = "received_date_time")
    private LocalDateTime receivedDateTime = null;
    @Column(name = "shipment_price")
    private double shipmentPrice = 0.0;
}

File: client/mobile/Sudoo/app/src/main/java/com/sudo248/sudoo/domain/entity/order/Shipment.kt
package com.sudo248.sudoo.domain.entity.order

import java.time.LocalDateTime

data class Shipment(
    val shipmentType: ShipmentType = ShipmentType.STANDARD,
    val deliveryTime: Int = 0,
    val shipmentPrice: Double = 0.0,
    val receivedDateTime: LocalDateTime? = null,
)
